src/Features/Core/Portable/GenerateOverrides/GenerateOverridesWithDialogCodeAction.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.PickMembers;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.GenerateOverrides
+{
+    internal partial class GenerateOverridesCodeRefactoringProvider
+    {
+        private class GenerateOverridesWithDialogCodeAction : CodeActionWithOptions
+        {
+            private readonly GenerateOverridesCodeRefactoringProvider _service;
+            private readonly Document _document;
+            private readonly INamedTypeSymbol _containingType;
+            private readonly ImmutableArray<ISymbol> _viableMembers;
+            private readonly TextSpan _textSpan;
+
+            public GenerateOverridesWithDialogCodeAction(
+                GenerateOverridesCodeRefactoringProvider service,
+                Document document,
+                TextSpan textSpan,
+                INamedTypeSymbol containingType,
+                ImmutableArray<ISymbol> viableMembers)
+            {
+                _service = service;
+                _document = document;
+                _containingType = containingType;
+                _viableMembers = viableMembers;
+                _textSpan = textSpan;
+            }
+
+            public override object GetOptions(CancellationToken cancellationToken)
+            {
+                var service = _service._pickMembersService_forTestingPurposes ?? _document.Project.Solution.Workspace.Services.GetService<IPickMembersService>();
```

Comment:
So this is how we've generally 'injected' mocks into many of our code action tests.  That's not to say it's the right way to do things.  Just it's how we always do it.  If you want to walk me through how to add a one-off special type into the test workspace' export-provider, then i'm happy to follow htis all up with a refactoring to our code-action tests to use that pattern.

As you've probably seen, i've been doing tons of cleanup to code-actoin tests.  I'm happy to continue that even further (once you help me figure out the right way to do it) :)